### PR TITLE
gh-76912: Raise OSError from any failure in getpass.getuser()

### LINKED ISSUE #76912: getpass.getuser() raises an unspecified exceptions (ImportError, OSError, etc)
BPO | [32731](https://bugs.python.org/issue32731)
--- | :---
Nosy | @gpshead, @matrixise, @jarondl, @fangyi-zhou, @jacobtylerwalls
PRs | <li>python/cpython#5616</li><li>python/cpython#29739</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2018-01-31.18:02:32.028>
labels = ['type-bug', 'library', '3.9', '3.10', '3.11']
title = 'getpass.getuser() raises an unspecified exceptions (ImportError, OSError, etc)'
updated_at = <Date 2021-11-24.03:33:25.025>
user = 'https://github.com/gpshead'
```

bugs.python.org fields:
```python
activity = <Date 2021-11-24.03:33:25.025>
actor = 'jacobtylerwalls'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2018-01-31.18:02:32.028>
creator = 'gregory.p.smith'
dependencies = []
files = []
hgrepos = []
issue_num = 32731
keywords = ['patch']
message_count = 4.0
messages = ['311349', '311358', '312079', '316587']
nosy_count = 5.0
nosy_names = ['gregory.p.smith', 'matrixise', 'jarondl', 'fangyizhou', 'jacobtylerwalls']
pr_nums = ['5616', '29739']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue32731'
versions = ['Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
[bpo-32731](https://bugs.python.org/issue32731)

**Before**
If the initial lookup against environment variables fails, `ImportError` was raised on Windows, and, on other systems, if the secondary lookup against the password database fails, `KeyError` was raised. Documentation only said an "Exception" is raised.

**Now**
Errors are re-raised as `OSError` and documented.

https://bugs.python.org/issue32731

* Issue: gh-76912

Fixes #76912

### CODE REVIEW

**Review by MaxwellDupre — APPROVED** [review]
Looks ok and tests pass.

**Comment by arhadthedev** [comment]
@gpshead as an author of the parent issue.

File: Doc/library/getpass.rst [hunk 1]
```diff
@@ -46,7 +46,7 @@ The :mod:`getpass` module provides two functions:
    :envvar:`USER`, :envvar:`!LNAME` and :envvar:`USERNAME`, in order, and
    returns the value of the first one which is set to a non-empty string.  If
    none are set, the login name from the password database is returned on
-   systems which support the :mod:`pwd` module, otherwise, an exception is
-   raised.
+   systems which support the :mod:`pwd` module, otherwise, an :exc:`OSError`
+   is raised.
```
Comment (gpshead) [inline Doc/library/getpass.rst:50]: Lets add a versionchanged 3.13 marker mentioning that a variety of inconsistent exceptions other than just OSError were raised in previous versions.

File: Lib/getpass.py [hunk 2]
```diff
@@ -156,17 +156,25 @@ def getuser():
 
     First try various environment variables, then the password
     database.  This works on Windows as long as USERNAME is set.
+    Any failure to find a username raises OSError.
 
+    .. versionchanged:: 3.13
```
Comment (gpshead) [inline Lib/getpass.py:161]: Move this into the getpass.rst docs for the function.

For the docstring, just keep it simple and mention in the new sentence that older Python versions raised a variety of exceptions instead.

Comment (jacobtylerwalls) [reply Lib/getpass.py:161]: Thanks, I like that. If you prefer I preserve some of this detail in getpass.rst I could, but based on the other comment I just simplified in both places.

Ready for another look!

**Review by gpshead — APPROVED** [review]

**Pull request merged**